common/app/routes/challenges/utils.js

```diff
@@ -326,12 +326,29 @@ export function filterComingSoonBetaChallenge(
 }
 
 export function filterComingSoonBetaFromEntities(
-  { challenge: challengeMap, ...rest },
+  { challenge: challengeMap, block: blocks, ...rest },
   isDev = false
 ) {
   const filter = filterComingSoonBetaChallenge.bind(null, isDev);
   return {
     ...rest,
+    block: Object.keys(blocks)
+      .map(dashedName => {
+        const block = blocks[dashedName];
+        let challenges = block.challenges;
+        if (Array.isArray(challenges)) {
+          challenges = challenges
+            .map(dashedName => challengeMap[dashedName])
+            .filter(filter)
+            .map(challenge => challenge.dashedName);
+        }
+        block.challenges = challenges;
```

Comment:
You are mutating state here. A better (FP) way would be something like this:

```js
return {
  ...block,
  challenges: [ ...filteredChallenges ]
};
```
This is the ~same as `Object.assign({}, block, { challenges: filteredChallenges })` where an empty object has the properties of `block` copied in to it, and then we overwrite the prop `challenges` with our new filtered challenges.